invenio_rdm_records/records/systemfields/statistics.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 TU Wien.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Cached transient field for record statistics."""
+
+from invenio_records.systemfields import SystemField
+
+from ...stats.utils import get_record_stats
+
+
+class RecordStatistics(SystemField):
+    """Field for lazy fetching and caching (but not storing) of record statistics."""
+
+    def __init__(self, key=None):
+        """Constructor."""
+        super().__init__(key=key)
+
+    #
+    # Data descriptor methods (i.e. attribute access)
+    #
+    def __get__(self, record, owner=None):
+        """Get the persistent identifier."""
+        if record is None:
+            # returns the field itself.
+            return self
+
+        stats = record.get("stats", None)
+        if stats:
+            return stats
+
+        recid = record["id"]
+        parent_recid = record.parent["id"]
+        stats = get_record_stats(recid=recid, parent_recid=parent_recid)
+        record["stats"] = stats
+
+        return stats
+
+    def pre_commit(self, record, **kwargs):
+        """Ensure that the statistics stay transient."""
+        record.pop("stats", None)
+
+    def pre_dump(self, record, data, **kwargs):
+        """Do nothing in particular before a record is dumped."""
+        # TODO this could be an alternative to the search dumper extension
```

Comment:
what do you think :question: :grey_question: :question:

Reply:
I agree that the logic is a bit scattered at the moment, and I also don't know of any other place where we have a strong opinion/reasoning on `pre_dump` vs. search dumper extension... I guess the latter could be customized more easily in case someone wants to disable e.g. statistics?

On the other hand, I have a stronger feeling on having it done here, since I think `pre_dump` applies to both item/DB and list/Search results. It's very likely that I'm wrong, so not sure if it's worth going down this rabbit-hole.

Reply:
Yeah, unless I'm misunderstanding, that was my thinking as well... The system field's `pre_dump()` might be called from a number of spots that we would need to check exhaustively.
I'm just gonna update the comments here (system field) and at the search dumper extension to point at each other, and leave it at that for now.